### ISSUE TITLE
[Feature Request] Generate holidays dates instead of using ICS files

### ISSUE THREAD

**Issue by tswistak** [SimpleMobileTools/Simple-Calendar#1533]
Hi!

I've encountered that function for adding holidays automatically isn't working well. For example in Poland all holidays ends in 2020. I've seen that you are using ICS files for that with hard-coded holidays - maybe it would be good to change this approach? I was thinking about generating holidays automatically for a given year or period of time. 

I don't know if there is something like that for Java/Kotlin (I'm not into this environment), but for example in JS world there is a [date-holidays](https://github.com/commenthol/date-holidays) library that does the job of generating holidays without internet connection for nearly 200 countries and even acknowledges differences in holidays between regions in one country (e.g. different states in the USA can have some holidays different). Maybe there is something like this available to use in Android?

If there isn't anything like this for Java/Kotlin, maybe it would be good to take a look how this JS library handles generating dates. They have a lot of [YAML files](https://github.com/commenthol/date-holidays/tree/master/data) for each country with written down dates of holidays or ways to count holidays (e.g. `1st monday of March` or something like `easter 60` to annotate 60 days after Easter). They also [have ways to count moving holidays](https://github.com/commenthol/date-holidays-parser/tree/master/src) that depends on the Moon (like Easter, Arabic holidays etc.) or something different.

I think that this kind of feature would be a good addition to the calendar. It still would be simple (you don't need to cover all countries, can be the same set as there is already), but there wouldn't be a need to change ICS files after few years.

Thanks for a good work!

**Comment by eneller** [comment]
see #1978

**Comment by tswistak** [comment]
> see #1978

I understand that you wanted to advertise your issue, but it's really not that much related. My feature request is about getting rid of ICS files. 

BTW. Instead of spamming in every issue related to holidays, you could just link them in your issue.

**Comment by eneller** [comment]
True, I thought this was the best way of linking them, as my issue got closed instantly even though i believe there was some merit to it.

However, i do believe it is very much related because generating holiday lists and a calendar app are two different purposes that should be separated. Reliable, free holiday lists are a feature many could profit from, not only this app.
